### PR TITLE
Gateway submaster fix

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayOrganisationChecksController.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            var username = _contextAccessor.HttpContext.User.UserDisplayName();
+
+            _logger.LogInformation($"RoatpGatewayController-EvaluateTradingNamePage-SubmitGatewayPageAnswer - ApplicationId '{viewModel.ApplicationId}' - PageId '{viewModel.PageId}' - Status '{viewModel.Status}' - UserName '{username}' - Comments '{comments}'");
+            try
+            {
+                await _applyApiClient.SubmitGatewayPageAnswer(viewModel.ApplicationId, viewModel.PageId, viewModel.Status, username, comments);
+            }
+            catch (Exception ex)
+            {
+                // MFCMFC Shutter page? throw again?
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayOrganisationChecksController.cs]: Can this be resolved?

File: src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs [hunk 2]
```diff
@@ -71,13 +73,13 @@ public async Task<RoatpGatewayApplicationViewModel> Handle(GetApplicationOvervie
...
-                        new GatewaySection { SectionNumber = 1, PageId = "1-10", LinkTitle = "Legal name", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 2, PageId = "1-20", LinkTitle = "Trading name", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 3, PageId = "1-30", LinkTitle = "Organisation status", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 4, PageId = "1-40", LinkTitle = "Address", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 5, PageId = "1-50", LinkTitle = "ICO registration number", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 6, PageId = "1-60", LinkTitle = "Website address", HiddenText = "", Status = "" },
-                        new GatewaySection { SectionNumber = 7, PageId = "1-70", LinkTitle = "Organisation high risk", HiddenText = "", Status = "" }
+                        new GatewaySection { SectionNumber = 1, PageId = "1-10", UrlTag ="legal-name", LinkTitle = "Legal name", HiddenText = "", Status = "" },
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs]: Are we removing these page ids?

Comment (MichaelYoung1981) [reply src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs]: Out of interest, why do we need all of this config in the handler?

File: src/SFA.DAS.AdminService.Web/Infrastructure/IRoatpApplicationApiClient.cs [hunk 3]
```diff
@@ -42,8 +46,24 @@ public interface IRoatpApplicationApiClient
 
        
          Task<List<GatewayPageAnswerSummary>> GetGatewayPageAnswers(Guid applicationId);
+
+         //MFCMFC THIS NEEDS TO GO ONCE ALL TIDY UP IS DONE
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Infrastructure/IRoatpApplicationApiClient.cs]: Can this be resolved?

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOrganisationChecksOrchestrator.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            var ukrlpDetail = await _applyApiClient.GetUkrlpDetails(request.ApplicationId);
+
+                if (ukrlpDetail.ProviderAliases != null && ukrlpDetail.ProviderAliases.Count > 0)
+                {
+                    model.UkrlpTradingName = ukrlpDetail.ProviderAliases.First().Alias;
+                }
+
+
+            var tradingNameAndWebsitePage = await _qnaApiClient.GetPageBySectionNo(request.ApplicationId, 0, 1, RoatpQnaConstants.RoatpSections.Preamble.SectionId.ToString());
+            model.ApplyTradingName = tradingNameAndWebsitePage?.PageOfAnswers?.SelectMany(a => a.Answers)?.FirstOrDefault(a => a.QuestionId == RoatpQnaConstants.RoatpSections.Preamble.QuestionIds.TradingName)?.Value;
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOrganisationChecksOrchestrator.cs]: To be discussed.

File: src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/Handlers/Gateway/GetApplicationOverviewHandler.cs]: We need a DevCuddle to figure out what to do about all the config (which is now in the overview).  I'm tackling the overview Monday so I'll think on it then, but I reckon it'll drop out during that redev

File: src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayOrganisationChecksController.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+            _logger = logger;
+            _orchestrator = orchestrator;
+        }     
+
+        [HttpGet("/Roatp/Gateway/{applicationId}/Page/legal-name")]
+        public async Task<IActionResult> GetGatewayLegalNamePage(Guid applicationId, string pageId)
+        {
+            var username = _contextAccessor.HttpContext.User.UserDisplayName();
+            var viewModel = await _orchestrator.GetLegalNameViewModel(new GetLegalNameRequest(applicationId, username));
+            return View("~/Views/Roatp/Apply/Gateway/pages/LegalName.cshtml", viewModel);
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/Controllers/Roatp/Apply/RoatpGatewayOrganisationChecksController.cs]: Will have a look at that during next phase -ie Monday (lifting the remaining pages to our new pattern)

File: src/SFA.DAS.AdminService.Web/Infrastructure/IRoatpApplicationApiClient.cs (hunk 3 above)
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/Infrastructure/IRoatpApplicationApiClient.cs]: I'll be resolving this as last step to getting everything clean, probably Monday

File: src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOrganisationChecksOrchestrator.cs (hunk 4 above)
Comment (MarkFCain) [reply src/SFA.DAS.AdminService.Web/Services/Gateway/GatewayOrganisationChecksOrchestrator.cs]: Consider it discussed :)

**Pull request merged**